### PR TITLE
when image in imagerepository Spec.Image.Name is already specified, d…

### PR DESCRIPTION
**Pull request by rcerven**
…on't

add namespace if it is already there

[STONEBLD-2530](https://issues.redhat.com//browse/STONEBLD-2530)

### CODE REVIEW

File: controllers/imagerepository_controller_test.go [hunk 1]
```diff
@@ -563,6 +565,33 @@ var _ = Describe("Image repository controller", func() {
 			imageRepository := getImageRepository(resourceKey)
 			Expect(imageRepository.Spec.Image.Name).To(Equal(expectedImageName))
 		})
+
+		It("should create image repository with requested name, includes already namespace", func() {
```
Comment (mmorhun) [inline controllers/imagerepository_controller_test.go:569]: `... with requested name that already includes ...`

File: controllers/imagerepository_controller.go [hunk 2]
```diff
@@ -313,7 +313,11 @@ func (r *ImageRepositoryReconciler) ProvisionImageRepository(ctx context.Context
 			imageRepositoryName = imageRepository.Namespace + "/" + imageRepository.Name
 		}
 	} else {
-		imageRepositoryName = imageRepository.Namespace + "/" + imageRepository.Spec.Image.Name
+		if strings.HasPrefix(imageRepository.Spec.Image.Name, "/"+imageRepository.Namespace+"/") || strings.HasPrefix(imageRepository.Spec.Image.Name, imageRepository.Namespace+"/") {
```
Comment (mmorhun) [inline controllers/imagerepository_controller.go:316]: Can we trim `/` at the beginning of `imageRepository.Spec.Image.Name`, so the condition became more clear?

File: controllers/imagerepository_controller_test.go [hunk 3]
```diff
@@ -46,15 +46,17 @@ var _ = Describe("Image repository controller", func() {
...
+	})
+
 	Context("Image repository provision", func() {
 
 		BeforeEach(func() {
 			quay.ResetTestQuayClientToFails()
 		})
 
 		It("should prepare environment", func() {
-			createNamespace(defaultNamespace)
```
Comment (mmorhun) [inline controllers/imagerepository_controller_test.go:56]: What's wrong with this?

File: controllers/imagerepository_controller_test.go (hunk 1 above)
Comment (rcerven) [reply controllers/imagerepository_controller_test.go:569]: ok

File: controllers/imagerepository_controller.go (hunk 2 above)
Comment (rcerven) [reply controllers/imagerepository_controller.go:316]: ok

**Review by mmorhun — APPROVED** [review]

**Pull request merged**